x-pack/solutions/observability/plugins/slo/server/services/get_slo_health.ts

```diff
@@ -5,74 +5,164 @@
-      .filter((item) => !!sloById[item.sloId])
-      .map((item) => ({
-        sloId: item.sloId,
-        sloInstanceId: item.sloInstanceId,
-        sloRevision: sloById[item.sloId].revision,
-        sloName: sloById[item.sloId].name,
-      }));
+    const page = typeof params.page === 'number' && params.page >= 0 ? params.page : 0;
+    const perPage =
+      typeof params.perPage === 'number' && params.perPage >= 1 && params.perPage <= 100
+        ? params.perPage
+        : 100;
+
+    do {
+      const sloIdCompositeQueryResponse = await this.scopedClusterClient.asCurrentUser.search({
+        index: SUMMARY_INDEX_TEMPLATE_PATTERN,
+        size: 0,
+        aggs: {
+          sloIds: {
+            composite: {
+              after: afterKey as Record<string, FieldValue>,
+              size: ES_PAGESIZE_LIMIT,
+              sources: [
+                {
+                  sloId: { terms: { field: 'slo.id' } },
+                },
+                {
+                  sloInstanceId: { terms: { field: 'slo.instanceId' } },
+                },
+                {
+                  sloRevision: { terms: { field: 'slo.revision' } },
+                },
+                {
+                  sloName: { terms: { field: 'slo.name.keyword' } },
+                },
+              ],
+            },
+          },
+        },
+        ...(params.list?.length && {
+          query: {
+            bool: {
+              filter: [{ terms: { 'slo.id': params.list.map((slo) => slo.sloId) } }],
+            },
+          },
+        }),
+      });
+
+      afterKey = getAfterKey(sloIdCompositeQueryResponse.aggregations?.sloIds);
+
+      const buckets = (
+        sloIdCompositeQueryResponse.aggregations?.sloIds as {
+          buckets?: Array<{
+            key: { sloId: string; sloInstanceId: string; sloRevision: number; sloName: string };
+          }>;
+        }
+      )?.buckets;
+
+      if (buckets && buckets.length > 0) {
+        sloKeysFromES = sloKeysFromES.concat([
+          ...buckets.map((bucket) => {
+            return {
+              sloId: bucket.key.sloId,
+              sloInstanceId: bucket.key.sloInstanceId,
+              sloRevision: bucket.key.sloRevision,
+              sloName: bucket.key.sloName,
+            };
+          }),
+        ]);
+      }
+    } while (afterKey);
+
+    const filteredList = sloKeysFromES.map((item) => ({
+      sloId: item.sloId,
+      sloInstanceId: item.sloInstanceId,
+      sloRevision: item.sloRevision,
+      sloName: item.sloName,
+    }));
+
+    let count = 0;
+    let i = 0;
+    let transformStats: Dictionary<TransformGetTransformStatsTransformStats> = {};
+
+    do {
+      const tempTransformStats = await this.getTransformStats(i);
+      transformStats = { ...transformStats, ...tempTransformStats.data };
+      count = tempTransformStats.count;
+      i += ES_PAGESIZE_LIMIT;
+    } while (i <= count);
 
+    const transformStatsById = await this.getTransformStats(0);
```

Comment:
The code right above aggregates paged transform stats into the transformStats variable in the `do { .... }` loop, but then you call `this.getTransformStats(0);` to only use the first-page transform stats. Need to understand this part better.